p2p/socket_mocks.go

```diff
@@ -5,8 +5,10 @@ import (
 	"context"
 	stdio "io"
 	"net"
-	"reflect"
+	"sync"
```

Comment:
If this mock is for testing purposes, it needs to be better organized within the code so that it's not included in `go get`

Make a test file may include `*testing.T` param to further enforce testing only

Reply:
testutil seems like an appropriate dir

Reply:
awesome suggestions, thank you